### PR TITLE
Readability fix in `memory.pyx`

### PR DESCRIPTION
**Pull request by niboshi**
Converts private free functions to instance (or class) methods of relevant classes.

### CODE REVIEW

File: cupy/cuda/memory.pyx [hunk 1]
```diff
@@ -919,6 +759,61 @@ cdef class _Arena:
...
+                self._free[index] = free_list = set()
+        else:
+            free_list = set()
+            self._index.insert(self._index.begin() + index, bin_index)
+            self._flag.insert(self._flag.begin() + index, 0)
+            self._free.insert(index, free_list)
+        free_list.add(chunk)
+        self._flag[index] = 1
+
+    cdef bint remove_from_free_list(self, _Chunk chunk) except *:
```
Comment (niboshi) [inline cupy/cuda/memory.pyx]: Seems like `except *:` is not necessary?

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
The test failure is because of `numpy==1.18.0` does not happen with `1.17.0`

**Comment by emcastillo** [comment]
Fix was merged, Jenkins, test this please

**Pull request merged**